core/cache/application.go

```diff
@@ -28,18 +33,78 @@ type Application struct {
 	configHash string
 }
 
-func (m *Application) setDetails(details ApplicationChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.details = details
+func (a *Application) WatchFields(comparitors ...func(*ApplicationDelta) bool) *applicationFieldWatcher {
+	w := newApplicationFieldWatcher(comparitors)
+
+	unsub := a.hub.Subscribe(a.topic(applicationFieldChange), w.detailsChange)
+
+	w.tomb.Go(func() error {
+		<-w.tomb.Dying()
+		unsub()
+		return nil
+	})
+
+	return w
+}
+
+// TODO (manadart 2018-03-13) Should we change this up and down the call stack
+// so that pointers are passed instead of copies?
+func (a *Application) setDetails(details ApplicationChange) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	delta := &ApplicationDelta{old: a.details, new: details}
+	a.details = details
 
 	configHash, err := hash(details.Config)
 	if err != nil {
 		logger.Errorf("invariant error - application config should be yaml serializable and hashable, %v", err)
 		configHash = ""
 	}
-	if configHash != m.configHash {
-		m.configHash = configHash
+	if configHash != a.configHash {
+		a.configHash = configHash
 		// TODO: publish config change...
 	}
+
+	a.hub.Publish(a.topic(applicationFieldChange), delta)
+}
+
+// topic prefixes the input string with the application name.
+func (a *Application) topic(suffix string) string {
+	return a.details.Name + ":" + suffix
+}
+
+type applicationFieldWatcher struct {
+	*notifyWatcherBase
+
+	comparitors []func(*ApplicationDelta) bool
+}
+
+func newApplicationFieldWatcher(comparitors []func(*ApplicationDelta) bool) *applicationFieldWatcher {
```

Comment:
There is the possibility of multiple watchers watching the same fields. `Model` uses a `hashCache` to ameliorate a similar situation for config watchers, but I don't think it will be so common with _field_ watchers for non-model entities.